### PR TITLE
Update katsu to the v3.0 data model part 1

### PR DESCRIPTION
**Pull request by SonQBChau**
## What's new
- Move `primary_site` from `Donor` to `Primary Diagnosis`
- Add `submitter_treatment_id` to `Specimen`
- Move `[“pathological_tumour_staging_system, “pathological_stage_group”,
“pathological_t_category”, “pathological_n_category”, “pathological_m_category”]` from `Specimen` to `Primary Diagnosis`
- Remove `[“lymph_nodes_examined_status”, “lymph_nodes_examined_method”, “number_lymph_nodes_positive”]` from `Primary Diagnosis`
- Remove fields `[“line_of_treatment”, “treatment_setting”]` from `Treatment`
- New `Systemic Therapy` table
- Remove `Chemotherapy, Hormone Therapy, Immunotherapy`
- Remove `submitter_specimen_id` and add `[“surgery_reference_database”, “surgery_reference_identifier”` to `Surgery`
- Remove `[“recurrence_tumour_staging_system”, “recurrence_stage_group”, “recurrence_t_category”, “recurrence_n_category”, “recurrence_m_category”]` from `Follow Up`

### How to test
- No testing, just code review

### CODE REVIEW

**Review by mshadbolt — APPROVED** [review]
Overall looks good to me, mostly had a look at the models.

**Review by kcranston — APPROVED** [review]
Reviewing against the model documents, this looks good to me.

**Comment by OrdiNeu** [comment]
What are the scripts in chord_metadata_service/mohpackets/data/ used for? `data_converter.py` and `data_loader.py` should either be updated or removed

File: chord_metadata_service/mohpackets/apis/discovery.py [hunk 1]
```diff
@@ -104,23 +103,8 @@ def discover_sidebar_list(request):
...
-        Immunotherapy.objects.exclude(drug_name__isnull=True)
-        .values_list("drug_name", flat=True)
-        .order_by("drug_name")
-        .distinct()
-    )
-
-    # Drugs queryable for hormone therapy
-    hormone_therapy_drug_names = list(
-        HormoneTherapy.objects.exclude(drug_name__isnull=True)
+    systemictherapy_drug_names = list(
```
Comment (OrdiNeu) [inline chord_metadata_service/mohpackets/apis/discovery.py]: For readability, should be systemic_therapy
```suggestion
    systemic_therapy_drug_names = list(
```

File: chord_metadata_service/mohpackets/apis/discovery.py [hunk 2]
```diff
@@ -129,9 +113,7 @@ def discover_sidebar_list(request):
     result = {
         "treatment_types": TREATMENT_TYPE,
         "tumour_primary_sites": PRIMARY_SITE,
-        "chemotherapy_drug_names": chemotherapy_drug_names,
-        "immunotherapy_drug_names": immunotherapy_drug_names,
-        "hormone_therapy_drug_names": hormone_therapy_drug_names,
+        "drug_names": systemictherapy_drug_names,
```
Comment (OrdiNeu) [inline chord_metadata_service/mohpackets/apis/discovery.py]:
```suggestion
        "drug_names": systemic_therapy_drug_names,
```

File: chord_metadata_service/mohpackets/apis/ingestion.py [hunk 3]
```diff
@@ -94,11 +90,11 @@ def create_biomarkers(
     return create_instances(payload, Biomarker)
 
 
-@router.post("/chemotherapies/")
-def create_chemotherapies(
-    request, payload: List[ChemotherapyIngestSchema], response: HttpResponse
+@router.post("/systemictherapies/")
+def create_systemictherapies(
```
Comment (OrdiNeu) [inline chord_metadata_service/mohpackets/apis/ingestion.py]: If a similar function is `create_primary_diagnoses`, this should follow that convention.
```suggestion
def create_systemic_therapies(
```

File: chord_metadata_service/mohpackets/schemas/filter.py [hunk 4]
```diff
@@ -88,38 +86,24 @@ class TreatmentFilterSchema(FilterSchema):
...
     submitter_treatment_id: Optional[str] = Field(None)
     drug_reference_database: Optional[str] = Field(None)
     drug_name: Optional[str] = Field(None)
     drug_reference_identifier: Optional[str] = Field(None)
-    hormone_drug_dose_units: Optional[str] = Field(None)
+    drug_dose_units: Optional[str] = Field(None)
     prescribed_cumulative_drug_dose: Optional[int] = Field(None)
     actual_cumulative_drug_dose: Optional[int] = Field(None)
+    days_per_cycle: Optional[int] = Field(None)
+    number_of_cycles: Optional[int] = Field(None)
```
Comment (OrdiNeu) [inline chord_metadata_service/mohpackets/schemas/filter.py:108]: Missing systemic_therapy_type; it feels like the sort of thing we should be able to filter on.

**Review by OrdiNeu — CHANGES_REQUESTED** [review]

**Comment by SonQBChau** [comment]
> What are the scripts in chord_metadata_service/mohpackets/data/ used for? `data_converter.py` and `data_loader.py` should either be updated or removed

It will be updated after I merge with Marion's synthetic data

**Comment by SonQBChau** [comment]
change on systemic_therapy as requested

**Review by yavyx — CHANGES_REQUESTED** [review]
The treatmentTypeEnum still has the old chemo/hormone/immuno values
[See here](https://github.com/CanDIG/katsu/blob/sonchau/model_3_part_1/chord_metadata_service/mohpackets/docs/schema.yml#L4661)

**Comment by SonQBChau** [comment]
> The treatmentTypeEnum still has the old chemo/hormone/immuno values [See here](https://github.com/CanDIG/katsu/blob/sonchau/model_3_part_1/chord_metadata_service/mohpackets/docs/schema.yml#L4661)

This change has been addressed in [part 3 ](https://github.com/CanDIG/katsu/pull/248/files#diff-9c4dd410e6ffb078dc81b326de7ed069dbf467816073d1817bcfa92989d79f09L507), so it will be included in the final version

**Comment by SonQBChau** [comment]
I updated the enum

File: chord_metadata_service/mohpackets/docs/schema.yml [hunk 5]
```diff
@@ -2554,14 +1879,19 @@ components:
...
-          pattern: ^[A-Za-z0-9\-\._]{1,64}
+          pattern: ^[A-Za-z0-9\-\._]{1,64}$
           title: Submitter Treatment Id
           type: string
         surgeries:
           items:
             $ref: '#/components/schemas/NestedSurgerySchema'
           title: Surgeries
           type: array
+        systemictherapies:
```
Comment (mshadbolt) [inline chord_metadata_service/mohpackets/docs/schema.yml]:
```suggestion
        systemic_therapies:
```
I think it needs an underscore to be consistent with how we have done it for other schemas

**Review by OrdiNeu — APPROVED** [review]

**Pull request merged**